### PR TITLE
[Node CLI] Add a command that returns the stake pool address

### PR DESCRIPTION
**Pull request by movekevin**
### Description
We now have three type of stake pools an account can own:
1. Directly via the stake module
2. Via forming multiple staking contracts with different operators
3. Indirectly via vesting module

### Test Plan
Manual tests

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4503)

### CODE REVIEW

File: crates/aptos/src/node/mod.rs [hunk 1]
```diff
@@ -231,6 +235,59 @@ impl ValidatorNetworkAddressesArgs {
...
+    async fn execute(mut self) -> CliTypedResult<AccountAddress> {
+        let owner_address = self.owner_address;
+
+        let client = self.rest_options.client(&self.profile_options.profile)?;
+        let stake_pool_exists = client
+            .get_account_resource_bcs::<StakePool>(owner_address, "0x1::stake::StakePool")
+            .await
+            .is_ok();
+        let stake_pool_address = if stake_pool_exists {
+            owner_address
```
Comment (gregnazario) [inline crates/aptos/src/node/mod.rs:267]: This doesn't really tell you if the address is a delegated pool or the original, but I guess that works.

File: crates/aptos/src/node/mod.rs [hunk 2]
```diff
@@ -231,6 +235,59 @@ impl ValidatorNetworkAddressesArgs {
...
+            if staking_contract_store_exists {
+                default_stake_pool_address(owner_address, operator_address)
+            } else {
+                create_vesting_pool_address(
+                    owner_address,
+                    operator_address,
+                    self.employee_account_index.unwrap(),
+                    &[],
+                )
+            }
```
Comment (gregnazario) [inline crates/aptos/src/node/mod.rs:285]: Maybe we should just show all the staking pools if there is no operator address provided.

File: crates/aptos/src/node/mod.rs [hunk 3]
```diff
@@ -231,6 +235,59 @@ impl ValidatorNetworkAddressesArgs {
...
+    pub(crate) owner_address: AccountAddress,
+    #[clap(long)]
+    pub(crate) operator_address: Option<AccountAddress>,
+    #[clap(long)]
+    pub(crate) employee_account_index: Option<u64>,
+    #[clap(flatten)]
+    pub(crate) rest_options: RestOptions,
+    #[clap(flatten)]
+    pub(crate) profile_options: ProfileOptions,
+}
```
Comment (gregnazario) [inline crates/aptos/src/node/mod.rs:250]: These need documentation / AccountAddress inputs should use load_account_arg

**Review by gregnazario — APPROVED** [review]

**Review by sherry-x — APPROVED** [review]

File: crates/aptos/src/node/mod.rs (hunk 3 above)
Comment (movekevin) [reply crates/aptos/src/node/mod.rs]: Updated

File: crates/aptos/src/node/mod.rs (hunk 2 above)
Comment (movekevin) [reply crates/aptos/src/node/mod.rs:292]: That'd take more work as we'd need to parse the SimpleMap. I can do that in a separate PR later

**Pull request merged**